spec/features/roles_spec.rb

```diff
@@ -0,0 +1,137 @@
+require 'spec_helper'
+
+feature Role do
+  let(:conference) { create(:conference) }
+  let(:role_names) { Role.all.each.map(&:name) }
+
+  shared_examples 'should successfully edit' do |role_name, by_role_name|
+    let!(:role) { Role.find_by(name: role_name, resource: conference) }
+    let!(:by_role) { Role.find_by(name: by_role_name, resource: conference) }
+    let!(:user_to_sign_in) { create(:user, role_ids: [by_role.id]) }
+
+    before :each do
+      sign_in user_to_sign_in
+      visit admin_conference_roles_path(conference.short_title)
+    end
+
+    scenario "role #{role_name}", feature: true, js: true do
+      click_link('Edit', href: edit_admin_conference_role_path(conference.short_title, role_name))
+      fill_in 'role_description', with: 'changed description'
+      click_button 'Update Role'
+      role.reload
+
+      expect(flash).to eq("Successfully updated role #{role_name}")
+      expect(role.description).to eq('changed description')
+    end
+  end
+
+  shared_examples 'should successfully' do |role_name, by_role_name|
+    let!(:role) { Role.find_by(name: role_name, resource: conference) }
+    let!(:user_with_role) { create(:user, role_ids: [role.id]) }
+    let!(:by_role) { Role.find_by(name: by_role_name, resource: conference) }
+    let!(:user_to_sign_in) { create(:user, role_ids: [by_role.id]) }
+    let!(:user_with_no_role) { create :user }
+
+    before :each do
+      sign_in user_to_sign_in
+      visit admin_conference_roles_path(conference.short_title)
+    end
+
+    scenario "add role #{role_name}", feature: true, js: true do
+      click_link('Users', href: admin_conference_role_path(conference.short_title, role_name))
+
+      fill_in 'user_email', with: user_with_no_role.email
+      click_button 'Add'
+      user_with_no_role.reload
+
+      expect(user_with_no_role.has_role?(role.name, conference)).to eq true
+    end
+
+    scenario "edit role #{role_name}", feature: true, js: true do
```

Comment:
This is not edit role, this is remove user from a role (or remove role, from a user, as you have phrased it accordingly in `add role`)